src/main/java/org/example/FileManager.java

```diff
@@ -106,4 +106,35 @@ public static void writeVehicleToFile(Vehicle vehicle){
 
     }
 
+    public static void saveDealership(Dealership dealership){
+
+        List<Vehicle> vehicles = dealership.getInventory();
+
+        String filePath = "src/main/resources/inventory.csv";
+
+        try(BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))){
+
+            writer.write(dealership.getName() + "|" + dealership.getAddress() + "|" + dealership.getPhone());
+
+            writer.newLine();
+
+            for(Vehicle vehicle : vehicles){
+
+                writer.write(String.format("%d|%d|%s|%s|%s|%s|%d|$%.2f \n", vehicle.getVin(), vehicle.getYear(), vehicle.getMake(), vehicle.getModel(), vehicle.getVehicleType(), vehicle.getColor(), vehicle.getOdometer(), vehicle.getPrice()));
+
+                writer.newLine();
+
+                vehicles.add(vehicle);
```

Comment:
I may be wrong but I think this creates an infinite loop adding the vehicle you're using to the list you're looping through